Aztec/Classes/NSAttributedString/Conversions/AttributedStringSerializer.swift

```diff
@@ -108,19 +108,19 @@ class AttributedStringSerializer {
         if let representation = implicitRepresentation(for: element, inheriting: childAttributes) {
             content.append(representation)
         } else {
-            // if the element is of type '.a' and contains a '.img'
-            if element.isNodeType(.a) && element.children.count == 1 {
-                if let imgElement = element.children.first as? ElementNode, imgElement.isNodeType(.img) {
-
-                    // get the link URL and assign it to the image attachment
-                    if let linkText = element.stringValueForAttribute(named: "href") {
-                        let imgAttributes = self.attributes(for: imgElement, inheriting: attributes)
-                        if let attachment = imgAttributes[NSAttachmentAttributeName] as? ImageAttachment,
-                            let attributedString = implicitRepresentation(for: imgElement, inheriting: imgAttributes) {
-                            attachment.linkURL = URL(string: linkText)
-                            content.append(attributedString)
-                        }
-                    }
+            // if the node is of type '.a' with one child of type '.img', retrieve the '.img' node
+            if let imgElement = linkedImageElementFor(element) {
+                // get the link URL text
```

Comment:
We'd normally avoid explaining in comments what the code does, since the code itself is readable enough to describe it.

In cases where the code is not easy to read, it normally means some effort can be put towards making it more readable (like you did with the method `linkedImageElementFor()`).